enterprise/backend/src/metabase_enterprise/remote_sync/task/table_cleanup.clj

```diff
@@ -0,0 +1,57 @@
+(ns metabase-enterprise.remote-sync.task.table-cleanup
+  "Task to clean up old records from remote_sync_task table"
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.schedule.cron :as cron]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [java-time.api :as t]
+   [metabase.task.core :as task]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2])
+  (:import
+   (org.quartz DisallowConcurrentExecution)))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private cleanup-job-key
+  "Job key for the remote sync table cleanup task."
+  (jobs/key "metabase.task.remote-sync.table-cleanup.job"))
+
+(def ^:private cleanup-trigger-key
+  "Trigger key for the remote sync table cleanup task."
+  (triggers/key "metabase.task.remote-sync.table-cleanup.trigger"))
+
+(def ^:private retention-days
+  "Number of days of data to retain in the remote sync tables."
+  30)
+
+(defn- trim-remote-sync-tasks!
+  "Delete remote_sync_task records older than retention-days based on started_at timestamp."
+  []
+  (log/infof "Attempting to delete remote_sync_task records older than %d days." retention-days)
+  (let [cutoff-date (t/minus (t/offset-date-time) (t/days retention-days))
+        deleted-count (t2/delete! :model/RemoteSyncTask {:where [:< :started_at cutoff-date]})]
+    (log/infof "Deleted %d remote_sync_task records. Cleanup successful." deleted-count)
+    deleted-count))
+
+(defn- trim-tables!
```

Comment:
why have 2 functions for this when they do the same thing?

Reply:
Need to refactor there used to be two tables that needed to be truncated.